<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Comparing LIONESS Regulatory Networks using limma • netZooR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/all.min.css" integrity="sha256-PbSmjxuVAzJ6FPvNYsrXygfGhNJYyZ2GktDbkMBqQZg=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/v4-shims.min.css" integrity="sha256-A6jcAdwFD48VMjlI3GDxUd+eCQa7/KWy6G9oe/ovaPA=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Comparing LIONESS Regulatory Networks using limma">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">netZooR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.7.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ALPACA_MaghaP.html">Detecting differential modules using ALPACA</a>
    </li>
    <li>
      <a href="../articles/ApplicationinGTExData_DeborahW.html">Building PANDA and LIONESS Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
    <li>
      <a href="../articles/ApplicationwithTBdataset_TianW.html">Application with TB dataset</a>
    </li>
    <li>
      <a href="../articles/MONSTER_DanielS.html">An Introduction to the MONSTER Package</a>
    </li>
    <li>
      <a href="../articles/SAMBAR_MariekeK.html">SAMBAR: Subtyping Agglomerated Mutations By Annotation Relations</a>
    </li>
    <li>
      <a href="../articles/lioness_limma_CamilaLR.html">Comparing LIONESS Regulatory Networks using limma</a>
    </li>
    <li>
      <a href="../articles/pandaR_DanielS.html">An Introduction to the pandaR Package</a>
    </li>
    <li>
      <a href="../articles/panda_CamilaLR.html">Building PANDA Regulatory Networks from GTEx Gene Expression Data in R</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Comparing LIONESS Regulatory Networks using limma</h1>
                        <h4 class="author">Camila Lopes-Ramos</h4>
            
            <h4 class="date">2019-12-20</h4>
      
      
      <div class="hidden name"><code>lioness_limma_CamilaLR.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>LIONESS (Linear Interpolation to Obtain Network Estimates for Single Samples) is a method for creating sample-specific networks. When applied to a PANDA regulatory network, the result is a set of gene regulatory networks, one for each sample in the gene expression dataset. More information on LIONESS can be found in the published paper: <a href="https://doi.org/10.1016/j.isci.2019.03.021" class="uri">https://doi.org/10.1016/j.isci.2019.03.021</a></p>
<p>In this vignette, we will compare LIONESS regulatory networks from 207 females and 238 males with colon cancer using RNA-Seq data from TCGA. We will compare the edge weights between females and males using a linear regression model and correcting for the covariates age, race, and disease stage, as available in the limma package. We will also compare the gene’s in-degree (defined as the sum of the gene’s incoming edge weights from all TFs in the network). Finally, we will perform gene set enrichment analysis to find the pathways enriched for genes differentially targeted by sex in colon cancer.</p>
<div id="install-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#install-packages" class="anchor"></a>Install packages</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>))
    <span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)
BiocManager<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"limma"</span>)
BiocManager<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"fgsea"</span>)
BiocManager<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"Biobase"</span>)
<span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"ggplot2"</span>)
<span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"igraph"</span>)</code></pre></div>
</div>
<div id="load-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(limma)
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(fgsea)
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ggplot2)
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(Biobase)
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(igraph)</code></pre></div>
</div>
</div>
<div id="load-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#load-the-data" class="anchor"></a>Load the data</h1>
<p>For the purposes of demonstrating the workflow, we will load only a subet of LIONESS networks. Our subset shows the edge weights for 50,000 edges (rows) by 445 samples (columns). Let’s take a look at the networks:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load a subset of LIONESS networks</span>
lioness &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"data/lioness_coloncancer_subset.txt"</span>,<span class="dt">stringsAsFactors =</span> F, <span class="dt">check.names =</span> F)
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(lioness[,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>])
<span class="co"># Add row names as "TF_gene" and remove columns 1 and 2 with TF and gene name</span>
<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(lioness) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(lioness, <span class="dv">1</span>, <span class="cf">function</span>(x){
  <span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(x[<span class="dv">1</span>], x[<span class="dv">2</span>], <span class="dt">sep=</span><span class="st">"_"</span>)
})
<span class="co"># Remove TF and gene columns</span>
lioness &lt;-<span class="st"> </span>lioness[,<span class="op">-</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)]
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(lioness[,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>])

<span class="co"># Load the complete gene in-degree (sum of all edge weights for each gene) and clinical data stored as an expression set </span>
obj &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/get.html">get</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="st">"data/inDegree_allEdges_coloncancer.rdata"</span>, <span class="dt">sep=</span><span class="st">"/"</span>)))
<span class="co"># Clinical information</span>
<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(obj)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">30</span><span class="op">:</span><span class="dv">35</span>]</code></pre></div>
</div>
<div id="compare-the-edge-weights" class="section level1">
<h1 class="hasAnchor">
<a href="#compare-the-edge-weights" class="anchor"></a>Compare the edge weights</h1>
<p>Compare the edge weights between males and females using linear regression model (limma package) and adjusting for covariates: stage, age, race.</p>
<div id="run-limma" class="section level2">
<h2 class="hasAnchor">
<a href="#run-limma" class="anchor"></a>Run limma</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Define the covariates</span>
gender &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(obj)<span class="op">$</span>gender),<span class="dt">levels=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"MALE"</span>,<span class="st">"FEMALE"</span>))

stage &lt;-<span class="st"> </span>(<span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(obj)<span class="op">$</span>uicc_stage))
stage[<span class="kw"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(stage))] &lt;-<span class="st"> "NA"</span>    
stage &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(stage)

race &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(obj)<span class="op">$</span>race)
race[<span class="kw"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(race))] &lt;-<span class="st"> "NA"</span>
race &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span>(race)

age &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(obj)<span class="op">$</span>age_at_initial_pathologic_diagnosis)
age[<span class="kw"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(age))] &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(age,<span class="dt">na.rm=</span><span class="ot">TRUE</span>)

design =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span>(<span class="op">~</span><span class="st"> </span>stage <span class="op">+</span><span class="st"> </span>race <span class="op">+</span><span class="st"> </span>age <span class="op">+</span><span class="st"> </span>gender)

<span class="co"># Run limma</span>
fitGood =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(lioness),design)
fitGood =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span>(fitGood)
tb =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span>(fitGood,<span class="dt">coef=</span><span class="st">"genderFEMALE"</span>,<span class="dt">number=</span><span class="ot">Inf</span>)
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(tb)</code></pre></div>
</div>
<div id="visualize-the-top-edges-with-differential-weights-by-sex" class="section level2">
<h2 class="hasAnchor">
<a href="#visualize-the-top-edges-with-differential-weights-by-sex" class="anchor"></a>Visualize the top edges with differential weights by sex</h2>
<p>We select the top 50 edges with differential edge weights by sex and convert them into an igraph graph.data.frame object for visualization. We color edges red if they have higher coefficients in the female group, and blue if they have higher coefficients in the female group.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">toptable_edges &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span>(tb), <span class="st">"_"</span>))),<span class="dv">2</span>))
z &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/limma/man/cbind.html">cbind</a></span>(toptable_edges[<span class="dv">1</span><span class="op">:</span><span class="dv">50</span>,], tb<span class="op">$</span>logFC[<span class="dv">1</span><span class="op">:</span><span class="dv">50</span>])
g &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/igraph/man/graph_from_data_frame.html">graph.data.frame</a></span>(z, <span class="dt">directed=</span><span class="ot">FALSE</span>)
<span class="kw"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(g)<span class="op">$</span>weight &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(z[,<span class="dv">3</span>])
<span class="kw"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(g)<span class="op">$</span>color[<span class="kw"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(g)<span class="op">$</span>weight<span class="op">&lt;</span><span class="dv">0</span>] &lt;-<span class="st"> "blue"</span>
<span class="kw"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(g)<span class="op">$</span>color[<span class="kw"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(g)<span class="op">$</span>weight<span class="op">&gt;</span><span class="dv">0</span>] &lt;-<span class="st"> "red"</span>
<span class="kw"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(g)<span class="op">$</span>weight &lt;-<span class="st"> </span><span class="dv">1</span>

<span class="kw"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="dt">mar=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))
<span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(g, <span class="dt">vertex.label.cex=</span><span class="fl">0.7</span>, <span class="dt">vertex.size=</span><span class="dv">10</span>,  <span class="dt">vertex.label.font=</span><span class="dv">3</span>, <span class="dt">edge.width=</span><span class="dv">5</span><span class="op">*</span>(<span class="kw"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(z[,<span class="dv">3</span>]))))</code></pre></div>
</div>
</div>
<div id="compare-genes-in-degree" class="section level1">
<h1 class="hasAnchor">
<a href="#compare-genes-in-degree" class="anchor"></a>Compare gene’s in-degree</h1>
<p>Compare the gene’s in-degree between males and females using linear regression model (limma package) and adjusting for covariates: stage, age, race.</p>
<div id="run-limma-1" class="section level2">
<h2 class="hasAnchor">
<a href="#run-limma-1" class="anchor"></a>Run limma</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">indegree &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/Biobase/man/assayData.html">assayData</a></span>(obj)[[<span class="st">"quantile"</span>]]
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(indegree[,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>])
<span class="co"># Use the same design matrix as before</span>
fitGood =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span>(indegree,design)
fitGood =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span>(fitGood)
tb_degree =<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span>(fitGood,<span class="dt">coef=</span><span class="st">"genderFEMALE"</span>,<span class="dt">number=</span><span class="ot">Inf</span>)
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(tb_degree)
<span class="co"># Save gene ranks</span>
indegree_rank &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span>(<span class="dt">object=</span>tb_degree[,<span class="st">"t"</span>], <span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(tb_degree))</code></pre></div>
</div>
</div>
<div id="gene-set-enrichment-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#gene-set-enrichment-analysis" class="anchor"></a>Gene Set Enrichment Analysis</h1>
<p>Well will use the fgsea package to perform gene set enrichment analysis. We need to point to a ranked gene list (for example the gene in-degree statistical difference (t value) between males and females), and a list of gene sets (or signatures) in gmt format to test for enrichment. The gene sets can be downloaded from MSigDB: <a href="http://software.broadinstitute.org/gsea/msigdb" class="uri">http://software.broadinstitute.org/gsea/msigdb</a> Same gene annotation should be used in the ranked gene list and gene sets.</p>
<div id="run-fgsea" class="section level2">
<h2 class="hasAnchor">
<a href="#run-fgsea" class="anchor"></a>Run fgsea</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gmt.file &lt;-<span class="st"> "data/c2.cp.kegg.v7.0.symbols.gmt"</span>
pathways &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/fgsea/man/gmtPathways.html">gmtPathways</a></span>(gmt.file)
fgseaRes &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/fgsea/man/fgsea.html">fgsea</a></span>(pathways, indegree_rank, <span class="dt">minSize=</span><span class="dv">15</span>, <span class="dt">maxSize=</span><span class="dv">500</span>, <span class="dt">nperm=</span><span class="dv">1000</span>)
<span class="kw"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(fgseaRes)
<span class="co"># Subset to pathways with FDR &lt; 0.05</span>
sig &lt;-<span class="st"> </span>fgseaRes[fgseaRes<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>,]
<span class="co"># Top 10 pathways enriched in females</span>
sig<span class="op">$</span>pathway[sig<span class="op">$</span>NES <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>][<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]
<span class="co"># Top 10 pathways enriched in males</span>
sig<span class="op">$</span>pathway[sig<span class="op">$</span>NES <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>][<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]</code></pre></div>
</div>
<div id="bubble-plot-of-differentially-targeted-pathways" class="section level2">
<h2 class="hasAnchor">
<a href="#bubble-plot-of-differentially-targeted-pathways" class="anchor"></a>Bubble plot of differentially targeted pathways</h2>
<p>Bubble plot of gene sets on y-axis and adjusted p-value (padj) on x-axis. Bubble size indicates the number of genes in each gene set, and bubble color indicates the normalized enrichment score (NES). Blue is for negative NES (enrichment of higher targeted genes in males), and red is for positive NES (enrichment of higher targeted genes in females).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dat &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(fgseaRes)

<span class="co"># Settings</span>
fdrcut &lt;-<span class="st"> </span><span class="fl">0.05</span> <span class="co"># FDR cut-off to use as output for significant signatures</span>
dencol_neg &lt;-<span class="st"> "blue"</span> <span class="co"># bubble plot color for negative ES</span>
dencol_pos &lt;-<span class="st"> "red"</span> <span class="co"># bubble plot color for positive ES</span>
signnamelength &lt;-<span class="st"> </span><span class="dv">4</span> <span class="co"># set to remove prefix from signature names (2 for "GO", 4 for "KEGG", 8 for "REACTOME")</span>
asp &lt;-<span class="st"> </span><span class="dv">3</span> <span class="co"># aspect ratio of bubble plot</span>
charcut &lt;-<span class="st"> </span><span class="dv">100</span> <span class="co"># cut signature name in heatmap to this nr of characters</span>

<span class="co"># Make signature names more readable</span>
a &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(dat<span class="op">$</span>pathway) <span class="co"># 'a' is a great variable name to substitute row names with something more readable</span>
<span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(a)){
  a[j] &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/substr.html">substr</a></span>(a[j], signnamelength<span class="op">+</span><span class="dv">2</span>, <span class="kw"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span>(a[j]))
}
a &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/chartr.html">tolower</a></span>(a) <span class="co"># convert to lower case (you may want to comment this out, it really depends on what signatures you are looking at, c6 signatures contain gene names, and converting those to lower case may be confusing)</span>
<span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(a)){
  <span class="cf">if</span>(<span class="kw"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span>(a[j])<span class="op">&gt;</span>charcut) { a[j] &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/substr.html">substr</a></span>(a[j], <span class="dv">1</span>, charcut), <span class="st">"..."</span>, <span class="dt">sep=</span><span class="st">" "</span>)}
} <span class="co"># cut signature names that have more characters than charcut, and add "..."</span>
a &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">"_"</span>, <span class="st">" "</span>, a)
dat<span class="op">$</span>NAME &lt;-<span class="st"> </span>a

<span class="co"># Determine what signatures to plot (based on FDR cut)</span>
dat2 &lt;-<span class="st"> </span>dat[dat[,<span class="st">"padj"</span>]<span class="op">&lt;</span>fdrcut,]
dat2 &lt;-<span class="st"> </span>dat2[<span class="kw"><a href="https://rdrr.io/r/base/order.html">order</a></span>(dat2[,<span class="st">"padj"</span>]),] 
dat2<span class="op">$</span>signature &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/factor.html">factor</a></span>(dat2<span class="op">$</span>NAME, <span class="kw"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(dat2<span class="op">$</span>NAME)))
<span class="co"># Determine what labels to color</span>
sign_neg &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/which.html">which</a></span>(dat2[,<span class="st">"NES"</span>]<span class="op">&lt;</span><span class="dv">0</span>)
sign_pos &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/which.html">which</a></span>(dat2[,<span class="st">"NES"</span>]<span class="op">&gt;</span><span class="dv">0</span>)
<span class="co"># Color labels</span>
signcol &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="ot">NA</span>, <span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(dat2<span class="op">$</span>signature))
signcol[sign_neg] &lt;-<span class="st"> </span>dencol_neg <span class="co"># text color of negative signatures</span>
signcol[sign_pos] &lt;-<span class="st"> </span>dencol_pos <span class="co"># text color of positive signatures</span>
signcol &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(signcol) <span class="co"># need to revert vector of colors, because ggplot starts plotting these from below</span>

<span class="co"># Plot bubble plot</span>
g&lt;-<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(dat2, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x=</span>padj,<span class="dt">y=</span>signature,<span class="dt">size=</span>size))
g<span class="op">+</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">fill=</span>NES), <span class="dt">shape=</span><span class="dv">21</span>, <span class="dt">colour=</span><span class="st">"white"</span>)<span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>()<span class="op">+</span><span class="st"> </span><span class="co"># white background, needs to be placed before the "signcol" line</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span>(<span class="dv">0</span>,fdrcut)<span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html">scale_size_area</a></span>(<span class="dt">max_size=</span><span class="dv">10</span>,<span class="dt">guide=</span><span class="st">"none"</span>)<span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html">scale_fill_gradient2</a></span>(<span class="dt">low=</span>dencol_neg, <span class="dt">high=</span>dencol_pos)<span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">axis.text.y =</span> <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="dt">colour=</span>signcol))<span class="op">+</span>
<span class="st">  </span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="dt">aspect.ratio=</span>asp, <span class="dt">axis.title.y=</span><span class="kw"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span>()) <span class="co"># test aspect.ratio</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#install-packages">Install packages</a></li>
      <li><a href="#load-packages">Load packages</a></li>
      </ul>
</li>
      <li><a href="#load-the-data">Load the data</a></li>
      <li>
<a href="#compare-the-edge-weights">Compare the edge weights</a><ul class="nav nav-pills nav-stacked">
<li><a href="#run-limma">Run limma</a></li>
      <li><a href="#visualize-the-top-edges-with-differential-weights-by-sex">Visualize the top edges with differential weights by sex</a></li>
      </ul>
</li>
      <li>
<a href="#compare-genes-in-degree">Compare gene’s in-degree</a><ul class="nav nav-pills nav-stacked">
<li><a href="#run-limma-1">Run limma</a></li>
      </ul>
</li>
      <li>
<a href="#gene-set-enrichment-analysis">Gene Set Enrichment Analysis</a><ul class="nav nav-pills nav-stacked">
<li><a href="#run-fgsea">Run fgsea</a></li>
      <li><a href="#bubble-plot-of-differentially-targeted-pathways">Bubble plot of differentially targeted pathways</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Tian Wang, Marouen Ben Guebila, John Platig, Marieke Kuijjer, Magha Padi, Rebekka Burkholz.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
